src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -498,50 +545,83 @@ private async Task<bool> OnSetVariableValue(MessageId id, int scopeId, string va
             var varToSetValue = varIds.FirstOrDefault(v => v.Name == varName);
             if (varToSetValue == null)
                 return false;
-            Result res = await SendMonoCommand(id, MonoCommands.SetVariableValue(scopeId, varToSetValue.Index, varName, varValue["value"].Value<string>()), token);
-            if (res.IsOk)
+            var res = await sdbHelper.SetVariableValue(id, ctx.ThreadId, scopeId, varToSetValue.Index, varValue["value"].Value<string>(), token);
+            if (res)
                 SendResponse(id, Result.Ok(new JObject()), token);
             else
                 SendResponse(id, Result.Err($"Unable to set '{varValue["value"].Value<string>()}' to variable '{varName}'"), token);
             return true;
         }
 
-        private async Task<Result> RuntimeGetProperties(MessageId id, DotnetObjectId objectId, JToken args, CancellationToken token)
+        internal async Task<Result> RuntimeGetProperties(SessionId id, DotnetObjectId objectId, JToken args, CancellationToken token)
```

Comment:
It feels a little odd to return a result from something that doesn't start with a message.  It looks like this was needed  to make it callable from MemberReferenceResolver that then unpacks the result?

Reply:
Changed.